src/components/heading/heading.test.tsx

```diff
@@ -0,0 +1,239 @@
+    "https://www.warnerbros.com/movies/heat"
+  );
+});
+
+test("focuses the back link, when `backLink` is a href on mouse down", () => {
+  render(
+    <Heading title="foo" backLink="https://www.warnerbros.com/movies/heat" />
+  );
+
+  const backLink = screen.getByRole("link", { name: "Back" });
+  fireEvent.mouseDown(backLink);
+
+  expect(backLink).toBeFocused();
+});
+
+test("renders a back button, when `backLink` prop is a function", () => {
+  const backLinkFunction = jest.fn();
+  render(<Heading title="foo" backLink={backLinkFunction} />);
+
+  const backLink = screen.getByRole("button", { name: "Back" });
+  expect(backLink).toBeVisible();
+});
+
+test("focuses the back link, when `backLink` prop is a function", () => {
+  const backLinkFunction = jest.fn();
+  render(<Heading title="foo" backLink={backLinkFunction} />);
+
+  const backLink = screen.getByRole("button", { name: "Back" });
+  fireEvent.mouseDown(backLink);
+
+  expect(backLink).toBeFocused();
+});
+
+/* styling test for coverage */
+test("alters the grid-column property on the subtitle node, when a back link/button is rendered", () => {
+  render(
+    <Heading
+      title="foo"
+      subheader="bar"
+      backLink="https://www.warnerbros.com/movies/heat"
+    />
+  );
+
+  const subtitle = screen.getByTestId("subtitle");
+
+  expect(subtitle).toHaveStyle("grid-column: 2");
+});
+
+/* styling test for coverage */
+test("alters the outline property on the back link/button, when the back link is focused and consumers opt out of the focus redesign", () => {
+  render(
+    <CarbonProvider focusRedesignOptOut>
+      <Heading title="foo" backLink="https://www.warnerbros.com/movies/heat" />
+    </CarbonProvider>
+  );
+
+  const backLink = screen.getByTestId("heading-back-button");
+  backLink.focus();
+
+  expect(backLink).toHaveStyleRule(
+    "outline",
+    "3px solid var(--colorsSemanticFocus500)",
+    { modifier: "button:focus" }
+  );
+});
+
+test("renders a divider when the `divider` prop is true", () => {
+  render(<Heading title="foo" divider />);
+
+  const separator = screen.getByRole("separator");
+  expect(separator).toBeVisible();
+});
+
+test("renders a separator when the `separator` prop is true", () => {
+  render(<Heading title="foo" divider separator />);
+
+  const separators = screen.getAllByRole("separator");
+
+  separators.forEach((separator) => {
+    expect(separator).toBeVisible();
+  });
+});
+
+test("renders `Pill` component nodes after the title, via the `pills` prop", () => {
+  const pillsNode = (
+    <>
+      <Pill data-role="pill">Pill one</Pill>
+      <Pill data-role="pill">Pill two</Pill>
+    </>
+  );
+
+  render(<Heading title="foo" pills={pillsNode} />);
+
+  const pillsContainer = screen.getByTestId("heading-pills-container");
+  const pills = screen.getAllByTestId("pill");
+
+  expect(pillsContainer).toHaveAttribute("data-element", "pills");
+  pills.forEach((pill) => {
+    expect(pill).toBeVisible();
+  });
+});
```

Comment:
**suggestion:** By doing this we have no way of knowing if they've both been rendered so instead of iterating through the pills, to make sure both are visible I would give them a different data-roles and assert they are both visible individually. Or alternatively, making sure `pills` has a length of two for e.g. 

The same thing could be said about the previous test for the divider/separator.

Reply:
I could add another check to see if 2 children are rendered etc. but not too sure using two data-roles is the way to go on this one. I'll see what the other reviewer says

Reply:
I agree with @nuria1110 here, and see that she pointed out the same problem that I've just pointed out with the divider/separator tests (but I've just typed those out so I'll leave them in now! 😅 ).

I think just asserting that the length is 2 is the right way here, rather than adding separate data-roles. But since these data-roles would be added just in the test rather than the component itself, there's no particular harm - and you're already adding the same data-role to each so it would just mean changing them to 2 different ones. (But overall a simple check on length is perfectly fine by me, as well as the visibility checks.)

Reply:
Yeah sorry I probably wasn't clear enough in my initial comment about the data-roles, basically what Robin said just doing this is what I meant, but yeah either or is fine my me
```
      <Pill data-role="pill1">Pill one</Pill>
      <Pill data-role="pill2">Pill two</Pill>
```